bundles/admin/admin-hierarchical-layerlist/views/adminLayerSettingsView.js

```diff
@@ -0,0 +1,1545 @@
+                        });
+                        warningDialog.show(me.instance.locale('admin.warningTitle'), warningMessage, [warningDialogOkBtn]);
+                        warningDialog.makeModal();
+                    }
+                }
+            },
+
+            /**
+             * Fetch wfs specific common data / sld styles
+             *
+             * @method __setupSldStyles
+             */
+            _setupSldStyles: function() {
+                var me = this,
+                    elem = me.$el;
+
+                if (me.sldStyles) {
+                    me._sldStylesUI(elem);
+                }
+
+                jQuery.ajax({
+                    type: 'POST',
+                    dataType: 'json',
+                    data: {},
+                    url: me.instance.sandbox.getAjaxUrl('SldStyles'),
+                    success: function(resp) {
+                        me.sldStyles = resp.sldStyles;
+                        me._sldStylesUI(elem);
+                    },
+                    error: function(jqXHR) {
+                        if (jqXHR.status !== 0) {
+                            me._showDialog(me.instance.locale('admin.errorTitle'), me.instance.locale('admin.sldStylesFetchError'));
+                        }
+                    }
+                });
+            },
+            /**
+             * Save new sld style
+             *
+             * @method _saveSldStyle
+             */
+            _saveSldStyle: function(sldName, sldXml) {
+                var me = this;
+
+
+                jQuery.ajax({
+                    type: 'POST',
+                    dataType: 'json',
+                    data: {
+                        name: sldName,
+                        xml: encodeURIComponent(sldXml)
+                    },
+                    url: me.instance.sandbox.getAjaxUrl('SldStyles'),
+                    success: function(resp) {
+                        me._showDialog("title", "New sld saved success / " + sldName);
+                        //Update UI
+                        me._sldStylesAppendUI(resp.id, sldName);
+
+                    },
+                    error: function(jqXHR) {
+                        if (jqXHR.status !== 0) {
+                            me._showDialog("title", "Save of new sld xml failed");
+                        }
+                    }
+                });
+            },
+            /**
+             * Creates sld style ui
+             * @method  _sldStylesUI
+             * @param   {Object}     elem jQuery elemen
+             * @private
+             */
+            _sldStylesUI: function(elem) {
+                var me = this,
+                    sldSele = elem.find('#add-layer-sld-style');
+
+                for (var i = 0; me.sldStyles != null && i < me.sldStyles.length; i++) {
+                    sldSele.append('<option value=' + me.sldStyles[i].id + ' >' + me.sldStyles[i].name + '</option>');
```

Comment:
Call the below method instead so the same thing is not done in two places. You can send the select element as optional third parameter for it.